lib/AppInfo/Application.php

```diff
@@ -39,6 +39,33 @@ class Application extends App implements IBootstrap {
 	public const DEFAULT_TRANSCRIPTION_MODEL_ID = 'whisper-1';
 	public const DEFAULT_IMAGE_SIZE = '1024x1024';
 	public const MAX_GENERATION_IDLE_TIME = 60 * 60 * 24 * 10;
+	public const DEFAULT_MAX_NUM_OF_TOKENS = 1000;
+	public const DEFAULT_QUOTA_PERIOD = 30;
+	
+	public const QUOTA_TYPE_TEXT = 0;
+	public const QUOTA_TYPE_IMAGE = 1;
+	public const QUOTA_TYPE_TRANSCRIPTION = 2;
+
+	
+	public const QUOTA_TYPES = [
+		self::QUOTA_TYPE_TEXT => 'text',
```

Comment:
This is a bit annoying, I feel, YMMV but I feel like simply always using a string as an identifier would simplify the logic a lot.

Reply:
I think why I did this is still carry over from C. I'm still not fully used to dealing with associative arrays and fall back on my old habits as this would have been the way to do it without associative arrays.

Reply:
Now types are numbers all the way and there's a separate translate method for the human readable unit/type in OpenAiApiService: https://github.com/nextcloud/integration_openai/pull/54/commits/44720e0aebfd3f549a6d9bd48a78df2872f26110